Source/DafnyLanguageServer/Program.cs

```diff
@@ -3,26 +3,31 @@
 using OmniSharp.Extensions.LanguageServer.Server;
 using Serilog;
 using System;
+using System.Diagnostics;
 using System.IO;
 using System.Reflection;
+using System.Threading;
 using System.Threading.Tasks;
 using OmniSharpLanguageServer = OmniSharp.Extensions.LanguageServer.Server.LanguageServer;
 
 namespace Microsoft.Dafny.LanguageServer {
+
   public class Program {
     private static async Task Main(string[] args) {
       var configuration = CreateConfiguration(args);
       InitializeLogger(configuration);
       try {
+        var cancelLanguageServer = new CancellationTokenSource();
         var server = await OmniSharpLanguageServer.From(
           options => options
             .WithInput(Console.OpenStandardInput())
             .WithOutput(Console.OpenStandardOutput())
             .ConfigureConfiguration(builder => builder.AddConfiguration(configuration))
             .ConfigureLogging(SetupLogging)
             .WithUnhandledExceptionHandler(LogException)
-            .WithDafnyLanguageServer(configuration)
+            .WithDafnyLanguageServer(configuration, cancelLanguageServer)
         );
+        cancelLanguageServer.Token.Register(() => server.ForcefulShutdown());
```

Comment:
I think it would be good to log something here.

Reply:
Added a log message, although in another location